apps/sharedtest/test/unit/advanced_telemetry_helper_test.js

```diff
@@ -0,0 +1,255 @@
+      this.sinon.stub(console, 'info');
+      // Suppress the error message
+      this.sinon.stub(console, 'warn');
+
+      ath = new AdvancedTelemetryHelper();
+      MockNavigatorSettings.mTriggerObservers(
+        AdvancedTelemetryHelper.TELEMETRY_ENABLED_KEY, { settingValue: true });
+      done();
+    });
+
+    teardown(function() {
+    });
+
+    test('should call console info once', function(done) {
+      ath.count('mymetric');
+      sinon.assert.calledOnce(console.info);
+      done();
+    });
+
+    test('should format the console message correctly', function(done) {
+      ath.count('mymetric');
+      sinon.assert.calledWith(console.info, 'telemetry|mymetric|1|4');
+      done();
+    });
+
+    test('should not write format the message if missing param',
+    function(done) {
+      ath.count();
+
+      sinon.assert.notCalled(console.info);
+      // Should call a warning message.
+      sinon.assert.calledOnce(console.warn);
+      done();
+    });
+  });
+
+  suite('linear metric formatting:', function() {
+    var ath;
+
+    setup(function(done) {
+      this.sinon.stub(console, 'info');
+      // Suppress the error message
+      this.sinon.stub(console, 'warn');
+
+      ath = new AdvancedTelemetryHelper();
+      MockNavigatorSettings.mTriggerObservers(
+        AdvancedTelemetryHelper.TELEMETRY_ENABLED_KEY, { settingValue: true });
+      done();
+    });
+
+    teardown(function() {
+    });
+
+    test('should call console info once', function(done) {
+      ath.linear('mymetric', 10, 1, 10000, 5);
+      sinon.assert.calledOnce(console.info);
+      done();
+    });
+
+    test('should format the message correctly with full params',
+    function(done) {
+      ath.linear('mymetric', 10, 1, 10000, 5);
+      sinon.assert.calledWith(console.info,
+        'telemetry|mymetric|10|1|1|10000|5');
+      done();
+    });
+
+    test('should format the message correctly without full params',
+    function(done) {
+      ath.linear('mymetric', 10);
+      sinon.assert.calledWith(console.info, 'telemetry|mymetric|10|1|');
+      done();
+    });
+
+    test('should not write format message if missing param', function(done) {
+      ath.linear();
+      sinon.assert.notCalled(console.info);
+      // Should call a warning message.
+      sinon.assert.calledOnce(console.warn);
+      done();
+    });
+
+    test('should not write format message if 3 params sent', function(done) {
+      ath.linear('mymetric', 10, 1);
+      sinon.assert.notCalled(console.info);
+      // Should call a warning message.
+      sinon.assert.calledOnce(console.warn);
+      done();
+    });
+
+    test('should not write format message if 4 params sent', function(done) {
+      ath.linear('mymetric', 10, 1, 1000);
+      sinon.assert.notCalled(console.info);
+      // Should call a warning message.
+      sinon.assert.calledOnce(console.warn);
+      done();
+    });
+
+    test('should handle 0 min value and correct to 1', function(done) {
```

Comment:
I'm thinking it's possible for zero to be a valid minimum value. For example, a metric like 'contact-search-results' that tracks the number of contacts found per search.